Update kdtree2.cpp

jrheller opened the pull request:
This fixes a bug that causes tree creation to fail for more than about one million points (for me, it failed when using 4004000 points).  Using a single precision variable to sum this many points causes loss of precision (i.e., eventually all but the most significant one or two digits of each new value added to the sum are lost).  The solution is to simply use a double precision float for the sum variable.

jrheller committed:
Update kdtree2.cpp

jmhodges commented:
I haven't had a chance to play with this code in many years. Mind providing a test?

jrheller commented:
This uses an array of 10,000,000 points with coordinates in the range (0.95,1.0). Creating the tree fails with infinite level of recursion using a single precision sum variable.

-----Original Message-----
From: Jeff Hodges [mailto:notifications@github.com]
Sent: Sunday, February 2, 2014 05:15 PM
To: 'jmhodges/kdtree2'
Cc: 'jrheller'
Subject: Re: [kdtree2] Update kdtree2.cpp (#2)

I haven't had a chance to play with this code in many years. Mind providing a test? 
—
Reply to this email directly or view it on GitHub.

jrheller commented:
One other thing: I had to add

#include <cstdio>

right after 

#include <iostream>

in kdtree2.cpp (for the printf calls) to get kdtree2.cpp to compile. 

-----Original Message-----
From: Jeff Hodges [mailto:notifications@github.com]
Sent: Sunday, February 2, 2014 05:15 PM
To: 'jmhodges/kdtree2'
Cc: 'jrheller'
Subject: Re: [kdtree2] Update kdtree2.cpp (#2)

I haven't had a chance to play with this code in many years. Mind providing a test? 
—
Reply to this email directly or view it on GitHub.

jmhodges commented:
You say "this uses". Did you mean to include another test file?

jrheller commented:
No, the 10,000,000 point array is created within the attached test file using random points with coordinate values in the range (0.95,1).

-----Original Message-----
From: Jeff Hodges [mailto:notifications@github.com]
Sent: Saturday, February 8, 2014 12:13 AM
To: 'jmhodges/kdtree2'
Cc: 'jrheller'
Subject: Re: [kdtree2] Update kdtree2.cpp (#2)

You say "this uses". Did you mean to include another test file?
—
Reply to this email directly or view it on GitHub.

jrheller commented:
I just modified the original test file "kdtree2_test.cpp" as follows:
1. Hard coded N as 10,000,000, dim as 3.
2. Changed the range of the randomly generated coordinates to (0.95,1) from (0,1)
3. Got rid of the timing code 

-----Original Message-----
From: Jeff Hodges [mailto:notifications@github.com]
Sent: Saturday, February 8, 2014 12:13 AM
To: 'jmhodges/kdtree2'
Cc: 'jrheller'
Subject: Re: [kdtree2] Update kdtree2.cpp (#2)

You say "this uses". Did you mean to include another test file?
—
Reply to this email directly or view it on GitHub.

jmhodges commented:
Sorry, I've not used this code in a long time and don't have the ability to code review it well.

In fact, this repo is going to be archived as I do not plan to maintain it futher.
